pkg/frontend/querymiddleware/codec_protobuf.go

```diff
@@ -278,6 +278,7 @@ func (f protobufFormatter) decodeVectorData(data *mimirpb.VectorData) (*Promethe
 	}
 
 	for i, sample := range data.Histograms {
+		sample := sample
```

Comment:
I've tried to find information on it, but ended up using ChatGPT to find some answers:
"However, keep in mind that if the slice is resized or reallocated, the memory address obtained through &slice[index] will no longer be valid, and using that address can lead to undefined behavior."

Did it BS me again?

Reply:
That isn't true. It will be still valid for referencing, but it may not keep pointing to the same slice position, which could be considered invalid if what you want is to modify the original slice.

[Consider this example](https://go.dev/play/p/NFFG5TVEUzO):

```go
package main

import "fmt"

func main() {
	slice := make([]int, 1, 1)
	first := &slice[0]
	*first = 1
	fmt.Printf("Before appending: %v\n", slice)
	slice = append(slice, 2)
	fmt.Printf("After appending: %v\n", slice)
	*first = 10
	fmt.Printf("After modifying: %v\n", slice)
}
```

Reply:
I see, we treat histograms pretty much immutable inputs for the query engine, so this should be ok then.
```suggestion

```

Reply:
Yes (as long as we do `&data.Histograms[i].Histogram,` that you mention below)